src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -148,7 +172,7 @@ public IPAddress(byte[] address)
         // and winsock only understands Int32.
         internal IPAddress(int newAddress)
         {
-            Address = (long)newAddress & 0x00000000FFFFFFFF;
+            _addressOrScopeId = (uint)newAddress & 0x00000000FFFFFFFF;
```

Comment:
Should we be asserting that newAddress >= 0 and <= uint.MaxValue?  I realize this code is preexisting, but It's not clear to me why the 0xFFFFFFFF is necessary here.  If we're expecting values outside of the uint range, why is it safe to truncate like this, and if we're not expecting such values, why is this needed?